frontend/src/hooks/use-available-hosts-alert.tsx

```diff
@@ -0,0 +1,50 @@
+/* Copyright Contributors to the Open Cluster Management project */
+
+import { getAgentsForSelection } from '@openshift-assisted/ui-lib/cim'
+import { useRecoilState, useSharedAtoms } from '../shared-recoil'
+import { Trans, useTranslation } from '../lib/acm-i18next'
+import React from 'react'
+import { NavigationPath } from '../NavigationPath'
+
+const useNoAvailableHostsAlert = (
+  controlPlaneType: 'hosted' | 'standalone'
+): { title: string; content: React.ReactNode } | undefined => {
+  const { t } = useTranslation()
+  const { agentsState, infraEnvironmentsState } = useSharedAtoms()
```

Comment:
can you instead of recoil start using useK8sWatch from ocp-sdk ?

Reply:
Using recoil means there's no need to handle loading state, it's take care of at load time
Since infra environments are already loaded for other components, it makes sense to me to use it

Reply:
You will need to add a loading state to the cards